### PR TITLE
Fix test_cc_starlark_api_additional_inputs when used with lld.

### PR DESCRIPTION
**Pull request by benjaminp**
Unlike the GNU linkers, lld does not insert an absolute symbol for every version in the version script into the symbol table. Check the version script is working by instead assigning a particular symbol a version.

### CODE REVIEW

File: src/test/shell/bazel/cc_integration_test.sh [hunk 1]
```diff
@@ -603,9 +603,8 @@ function test_cc_starlark_api_additional_inputs() {
   cat >> "$pkg"/BUILD << EOF
 cc_bin(
     name = "g",
-    srcs = ["e.cc"],
```
Comment (buildbreaker2021) [inline src/test/shell/bazel/cc_integration_test.sh:606]: Cannot the test pass with an already existing target? Why is this modification necessary?

File: src/test/shell/bazel/cc_integration_test.sh [hunk 2]
```diff
@@ -603,9 +603,8 @@ function test_cc_starlark_api_additional_inputs() {
...
-    data = [":f"],
-    linkstatic = 1,
```
Comment (buildbreaker2021) [inline src/test/shell/bazel/cc_integration_test.sh:608]: Cannot the test pass with an already existing target? Why is this change necessary?

File: src/test/shell/bazel/cc_integration_test.sh [hunk 3]
```diff
@@ -622,9 +621,7 @@ EOF
   bazel build --experimental_cc_skylark_api_enabled_packages=, --verbose_failures \
     //"$pkg":g  &>"$TEST_log" || fail "Build failed"
 
-  nm bazel-bin/"$pkg"/g  | grep VERS_42.0 || fail "VERS_42.0 not in binary"
-
-  bazel-bin/"$pkg"/g | grep a1a2bcddcb2a1a || fail "output is incorrect"
```
Comment (buildbreaker2021) [inline src/test/shell/bazel/cc_integration_test.sh:627]: What's the reason behind deleting this line? Can the be preserved?

**Comment by benjaminp** [comment]
Thank you for the review.

To hopefully answer both of your questions: The intention of this test appears to be to check that a linker script can be successfully passed using the `additional_inputs` flag to `cc_common.link`. The test was previous relying on the linker inserting a symbol from the version script into the binary regardless of whether any symbols were given that version. However, lld does not have this behavior. Therefore, I modified the test binary and linker script to assign a version to a specific symbol, `f`.

**Review by buildbreaker2021 — APPROVED** [review]
